src/remote/remote.go

```diff
@@ -594,7 +594,9 @@ func (c *Client) execute(tid int, target *core.BuildTarget, command *pb.Command,
 		}
 	}
 	// We didn't actually upload the inputs before, so we must do so now.
-	command, digest, err := c.uploadAction(target, isTest, false)
+	command, newDigest, err := c.uploadAction(target, isTest, false)
+	// Update the original digest to ensure it matches the uploaded action.
```

Comment:
How is this different from before? Are you deliberately mutating the passed-in digest for other things to observe it?

Reply:
Yes because there is a map storing the digests which is used later to pull data for a target, 'unstampedBuildActionDigests'. We need to get the digest out of this function. It is probably better to just return it, so I can do that instead.

Reply:
I don't think we should be updating the action digest. It should be the same so I think we should nail down why they're different. I have an idea though.